test/unit/Account.test.js

```diff
@@ -83,28 +103,57 @@ describe('`fetchAccountPage`', () => {
   });
 
   describe('"settings" content', () => {
-    it('should not make axios request', () => {
-      fetchAccountPage(renderMockReq('settings'), mockRes, mockResolve);
+    let getHomeLibrarySpy;
+    before(() => {
+      global.store = {
+        getState: () => ({
+          patron,
+        }),
+      };
+      getHomeLibrarySpy = sinon.spy(Account, 'getHomeLibrary');
+    });
 
+    it('should not make axios request', () => {
+      Account.fetchAccountPage(renderMockReq('settings'), mockRes, mockResolve);
       expect(axiosGet.notCalled).to.equal(true);
     });
+    xit('should call getHomeLibrary', () => {
+      // not working :/
+      expect(getHomeLibrarySpy.called).to.equal(true);
+    })
```

Comment:
This feels like the easiest way to share my suggestion for how you could stub this:

https://github.com/NYPL/discovery-front-end/commit/611c40a64dc2144922f9d91182bdbfee61809109

There are probably cleaner ways, but I like being really explicit about what fixture we're drawing in - and I like to keep stubbing and spying as close line-wise to the tests as possible.

Reply:
I've not done a lot with spies personally, but the challenge of adding a spy to a module function that exports in the way that `Account` currently does is a familiar problem with stubbing. I think the issue may be that you're attempting to spy on `Account.getHomeLibrary` but when you call `Account.fetchAccountPage`, it is calling `getHomeLibrary` internally, which is technically a different function than the one exposed via `Account.getHomeLibrary`. (I think [this person hit a similar issue](https://stackoverflow.com/questions/39868119/how-to-sinon-spy-module-export-utility-functions).) A way around that is to explicitly call `Account.getHomeLibrary` from within `Account.fetchAccountPage`, a technique summarized here:

https://github.com/NYPL/discovery-front-end/commit/595d7b8fbb2ff37acb0d11e44594cf7d3b04cd27

But I've never liked adapting code to accommodate testing, so you might find that too drastic a change. FWIW I think verifying a test via stubbing is sufficient in this case - no need to also spy.